src/components/input/InputWrapper.tsx

```diff
@@ -0,0 +1,22 @@
+import React, { Fragment, PropsWithChildren } from 'react';
+import { InputGroup } from '@chakra-ui/react';
+
+import { Props as InputProps } from '../input';
+
+type Props = {
+  size?: InputProps['size'];
+  shouldWrap?: boolean;
+};
+
+const InputWrapper = ({
+  children,
+  size,
+  shouldWrap,
+}: PropsWithChildren<Props>) =>
+  shouldWrap ? (
+    <InputGroup size={size}>{children}</InputGroup>
```

Comment:
This renders an extra `div` around the `input` and we don't want to make DOM more complex

Reply:
Personal take, conditionally wrapping the element makes it more complex as we will have a different structure of the element based on weather or not a left or right child element is present... It does reduce DOM nesting at the cost of higher complexity....

Reply:
In this instance, where we do render a lot of inputs on the filters and on the insertion form I think simplifying the DOM is worth it. We are already getting a lighthouse warning on the DOM complexity on the SRP and I don't want to explode it.

Reply:
Ah, you are referring to the excessive DOM size warning, was somewhere else when thinking about DOM complexity...

Simplifying a component to counteract a problem of the consumer feels like the wrong call. In other words, I have in mind those excessive DOM warnings are shown for 1000+(from the top of my head) levels of nestings so we have much more leverage in the project to counteract them if we intend to, add one more or less level of nesting to a specific component should not really matter for that.